assets/src/models/ladderVehiclePosition.ts

```diff
@@ -0,0 +1,156 @@
+import { LadderDirection } from "../components/ladder"
+import {
+  centerOfVehicleGroupY,
+  heightOfVehicleGroup,
+} from "../components/ladderVehicle"
+import { Vehicle } from "../skate"
+
+export enum VehicleDirection {
+  Up,
+  Down,
+}
+
+export interface LadderVehiclePosition {
+  vehicle: Vehicle
+  x: number
+  y: number
+  vehicleDirection: VehicleDirection
+  lane?: number
+}
+
+type Range = [number, number]
+
+export const vehicleDirectionOnLadder = (
+  vehicle: Vehicle,
+  ladderDirection: LadderDirection
+): VehicleDirection =>
+  (vehicle.direction_id === 1) ===
+  (ladderDirection === LadderDirection.ZeroToOne)
+    ? VehicleDirection.Down
+    : VehicleDirection.Up
+
+export const putIntoLanes = (
+  ladderVehiclePositions: LadderVehiclePosition[]
+): LadderVehiclePosition[] =>
+  ladderVehiclePositions
+    .slice()
+    .sort(byDirectionAndY)
+    .reduce(
+      (
+        previousLadderVehiclePositions: LadderVehiclePosition[],
+        ladderVehiclePosition: LadderVehiclePosition
+      ): LadderVehiclePosition[] => {
+        const overlappers = overlappingPreviousVehicles(
+          previousLadderVehiclePositions,
+          ladderVehiclePosition
+        )
+        const occupiedLanes: number[] = overlappers.reduce(
+          (lanes: number[], overlapper: LadderVehiclePosition) =>
+            overlapper.lane !== undefined ? [...lanes, overlapper.lane] : lanes,
+          []
+        )
+        const lane = firstOpenLane(occupiedLanes)
+
+        return [
+          {
+            ...ladderVehiclePosition,
+            lane,
+          },
+          ...previousLadderVehiclePositions,
+        ]
+      },
+      []
+    )
+
+export const xInLaneWithOffset = (ladderVehicleHorizontalOffset: number) => ({
+  x,
+  vehicleDirection,
+  lane,
+}: LadderVehiclePosition): number => {
+  const laneOffset = (lane || 0) * ladderVehicleHorizontalOffset
+  return vehicleDirection === VehicleDirection.Up
+    ? x + laneOffset
+    : x - laneOffset
```

Comment:
I feel like this could be simpler. Like, maybe `x` doesn't need to be part of `LadderVehiclePosition`, and you could do all the x calculations at once based on the `lane` and `vehicleDirection`.